### DISCUSSION TITLE
Modbus support for SE EV charger or EV charging via Inverter

### DISCUSSION THREAD

**Discussion by konstar** [WillCodeForCats/solaredge-modbus-multi#284] [Q&A] [answered]
Hi,

I'm reviewing the full installation guide for the SolarEdge EV Charger, and it looks like it supports Modbus over RS485. If the Inverter is already configured for Modbus over RS485 then it should be good to go.
https://knowledge-center.solaredge.com/sites/kc/files/smart-ev-charger-installation-guide-aus.pdf

Wondering if anyone has the SE EV Charger installed and it is discovered by solaredge-modbus-multi? What entities does it discover and what can you control through Home Assistant?

Thanks.

**Comment by WillCodeForCats** [comment]
There isn't anything for this device in modbus that I'm aware of.

**Comment by jovana** [comment]
I have ordered this Charger. I need to wait before it has been installed. If it has been installed I will check what is discovered by the SMM. But like @WillCodeForCats already said, don't have too much hope.

**Comment by FranciscoBigWings** [comment]
I have a SE inverter with integrated charger. How could I check if any information can be read or written by this integration? 

It could be cool to at least turn on/off the EV charger. The inverter with EV charger has a pushbutton that does exactly that. The fact that is a pushbutton and a not a switch makes me think that this feature is controlled by software.

The main problem that I see is that EV charger PCB is an extra PCB separate from the PCB inverter and maybe it doesn’t follow Sunspec

**Comment by WillCodeForCats** [comment, marked as answer]
See https://github.com/WillCodeForCats/solaredge-modbus-multi/discussions/285 about adding support for other devices.

**Comment by Largelos** [comment]
What charger do you have exact? 
SolarEdge Home EV Charger **SE-EVK22CRM-01** in fact is branded KEBA p30 charger, what can be monitored and managed without problem by Ethernet UDP interface and HA KEBA integration. [Here ](https://community.home-assistant.io/t/evse-keba-p30-integration/52571) is community discussion.

**Comment by FranciscoBigWings** [comment]
In my case, before SE released the type of charger you mentioned. The EV charger was part of the inverter

This is a video that shows the inverter with EV charger included. https://youtu.be/Ox7q0wVLjRM

Support from SE to the EV is null, I’m really disappointed (as many other users) and searching for a ways to have a better control of the EV charger through HA.Thanks for your interest in the matter
